test/ResourceNoManualCurrencyFormatting.test.js

```diff
@@ -0,0 +1,275 @@
+        const rule = new ResourceSourceChecker(findRuleDefinition("source-no-manual-currency-formatting"));
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "matcher.test",
+            sourceLocale: "en-US",
+            source: "It costs ${num}.",
+            pathName: "a/b/c.xliff"
+        });
+        const actual = rule.matchString({
+            source: resource.getSource(),
+            target: resource.getTarget(),
+            resource,
+            file: "a/b/c.xliff"
+        });
+        expect(actual).toBeTruthy();
+        expect(actual.length).toBe(1);
+
+        expect(actual[0].severity).toBe("error");
+        expect(actual[0].id).toBe("matcher.test");
+        expect(actual[0].description).toBe("Do not format currencies in English strings. Use a locale-sensitive number formatter and substitute the result of that into this string.");
+        expect(actual[0].highlight).toBe("Source: It costs <e0>${num}</e0>.");
+        expect(actual[0].pathName).toBe("a/b/c.xliff");
+    });
+
+    test("ResourceNoManualCurrencyFormatting", () => {
+        expect.assertions(2);
+
+        const rule = new ResourceSourceChecker(findRuleDefinition("source-no-manual-currency-formatting"));
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "matcher.test",
+            sourceLocale: "en-US",
+            source: "It costs {num}.",
+            pathName: "a/b/c.xliff"
+        });
+        const actual = rule.matchString({
+            source: resource.getSource(),
+            target: resource.getTarget(),
+            resource,
+            file: "a/b/c.xliff"
+        });
+        expect(!actual).toBeTruthy();
+    });
+
+    test("ResourceManualCurrencyFormattingWithWhitespace", () => {
+        expect.assertions(8);
+
+        const rule = new ResourceSourceChecker(findRuleDefinition("source-no-manual-currency-formatting"));
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "matcher.test",
+            sourceLocale: "en-US",
+            source: "It costs $ {num}.",
+            pathName: "a/b/c.xliff"
+        });
+        const actual = rule.matchString({
+            source: resource.getSource(),
+            target: resource.getTarget(),
+            resource,
+            file: "a/b/c.xliff"
+        });
+        expect(actual).toBeTruthy();
+        expect(actual.length).toBe(1);
+
+        expect(actual[0].severity).toBe("error");
+        expect(actual[0].id).toBe("matcher.test");
+        expect(actual[0].description).toBe("Do not format currencies in English strings. Use a locale-sensitive number formatter and substitute the result of that into this string.");
+        expect(actual[0].highlight).toBe("Source: It costs <e0>$ {num}</e0>.");
+        expect(actual[0].pathName).toBe("a/b/c.xliff");
+    });
+
+    test("ResourceManualCurrencyFormattingCrazyParamName", () => {
```

Comment:
Just wanted to mention that since we're using Jest now, maybe we could start using more "natural" names/descriptions for tests like

```js
describe("source-no-manual-currency-formatting") {
    it("should correctly handle crazy replacement parameter names") {
        // ...
    }
}
```

Reply:
fixed (sort of)

Reply:
Frankly, I personally prefer the convention where the suite mentions component under test and test name also indicates the expected result. Feel free to disregard if You'd rather not do it that way, it's a matter of taste after all.

Based off of your recent changes I'd do it more like:

* Rule to disallow manual currency formatting in Resources
    * should match formatting with $
    * should not match a replacement param with no manual currency formatting
    * should match formatting with $ and whitespace
    * should match formatting when there is a crazy replacement param name
    * should match formatting at the start of the string
    * should match formatting at the end of the string
    * should match formatting in variants of a Plural Resource
    * should match formatting in variants of an Array Resource

This could be further slimmed down to "matches / does not match", but to me the "should"s flow very naturally, as if I were describing it to someone
> So, over here you have a rule which is supposed to detect manual currency formatting. It should return a match when someone does the manual formatting by putting a dollar sign before a replacement parameter. It should also return a match when there's a space between the dollar and the param. However, it should not return a match when there is no dollar sign at all.

Reply:
It's sort of a philosophical difference. One style is "test when the conditions are X" whereas what you are advocating is "test for a certain result". Either way, the code works as is and the tests verify it properly. We can work out a philosophy to use for test names going forward, and we should discuss in our sync meetings.